lib/layers/data/repositories/vehicles_repo_impl.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:wot_statistic/layers/domain/entities/vehicles_data.dart';
+import 'package:wot_statistic/layers/domain/repositories/vehicles_repo.dart';
+
+class VehiclesRepoImpl implements VehiclesRepo {
+  final VehiclesLocalDataSource vehiclesLocalSource;
+  final RemoteDataSource remoteSource;
+
+  const VehiclesRepoImpl({
+    required this.vehiclesLocalSource,
+    required this.remoteSource,
+  });
+
+  UserData get signedUser {
+    final _signedUser = vehiclesLocalSource.getSignedUser();
+    if (_signedUser == null) throw Exception(S.current.SignedUserIsNotExist);
+    return _signedUser;
+  }
+
+  @override
+  Future<List<Vehicle>> fetchUserVehicles() async {
+    await _initVehiclesDatabase();
+    final UserVehiclesApi userVehicles = await remoteSource.fetchUserVehicles(
+      accountId: signedUser.id,
+      accessToken: signedUser.accessToken,
+    );
+    final List<int> tankIds = userVehicles.createListOfTankId();
+    final List<VehiclesDataTTC> vehiclesByIdFromDb =
+        await vehiclesLocalSource.fetchTTCByListOfIDs(tankIds);
+    final List<Vehicle> result = _createVehicleListFrom(
+        vehiclesByIdFromDb, userVehicles.vehicles.values.first);
+    return result;
+  }
+
+  List<Vehicle> _createVehicleListFrom(List<VehiclesDataTTC> vehiclesByIdFromDb,
+      List<UserVehicle> userVehicles) {
+    List<Vehicle> result = [];
+    for (var item in vehiclesByIdFromDb) {
+      result.add(_createVehicleFromTtcAndUser(
+          userVehicles.firstWhere((e) => e.tankId == item.tankId), item));
+    }
+    return result;
+  }
+
+  Vehicle _createVehicleFromTtcAndUser(
+    UserVehicle userVehicle,
+    VehiclesDataTTC vehiclesDataTTC,
+  ) =>
+      Vehicle(
+        markOfMastery: userVehicle.markOfMastery,
+        wins: userVehicle.tankStat.wins,
+        battles: userVehicle.tankStat.battles,
+        description: vehiclesDataTTC.description,
+        image: vehiclesDataTTC.images.bigIcon,
+        isPremium: vehiclesDataTTC.isPremium,
+        isGift: vehiclesDataTTC.isGift,
+        name: vehiclesDataTTC.name,
+        nation: vehiclesDataTTC.nation,
+        type: vehiclesDataTTC.type,
+        tier: vehiclesDataTTC.tier,
+      );
+
+  Future<void> _initVehiclesDatabase() async {
+    final String currentLng = vehiclesLocalSource.getCurrentLng();
+    final String vehiclesDbLng = vehiclesLocalSource.getVehiclesCurrentLng();
+    final int databaseTtcCount = vehiclesLocalSource.getVehiclesTTCCount();
+    final VehiclesDataMeta vehiclesDataMeta;
+    try {
+      vehiclesDataMeta = (await remoteSource.fetchVehiclesDatabase(
+        limit: 1,
+        pageNumber: 1,
+        language: currentLng,
+      ))
+          .meta;
+    } catch (e) {
+      throw Exception(S.current.CheckInternetConnection);
+    }
+    if (vehiclesDataMeta.total == databaseTtcCount &&
+        currentLng == vehiclesDbLng) return;
+    vehiclesLocalSource.setVehiclesCurrentLng(currentLng);
```

Comment:
I this this solution with language and saving count everywhere, with each new database you will need to duplicate this code, please think how you can avoid so much duplications